scripts/typemock/src/arg.ts

```diff
@@ -0,0 +1,298 @@
+/**
+ * Represents an argument condition used during verification.
+ */
+export class Arg {
+    private _validate: (value: any) => boolean;
+    private _message: string;
+    private _rest: boolean;
+
+    private constructor(condition: (value: any) => boolean, message: string, rest = false) {
+        this._validate = condition;
+        this._message = message;
+        this._rest = rest;
+    }
+
+    /**
+     * Allows any value.
+     */
+    public static any<T = any>(): T & Arg {

```

Comment:
Point of contention: I dislike lying to the typesystem like this just to say "this function takes either its original argument types or one of these special verifier indicator values". You're giving up internal consistency for the sake of a better feeling API. These results for `any` and `match` in no way have the shape of a `T`. I do understand why it's being done, it just feels very unfortunate. Not sure what the action here is; maybe just a note on how manipulating/mapping signatures in some way would be useful?